data-raw/YBFMP.R

```diff
@@ -12,55 +13,194 @@ Mode <- function(x) {
-            Tide=Mode(Tide),
-            Microcystis=Mode(Microcystis),
-            Notes=paste(na.omit(Notes), collapse="; "),
-            .groups="drop")%>%
-  mutate(across(where(is.numeric), ~if_else(is.nan(.x), NA_real_, .x)), # Replace NaN values with NAs
-         Notes=na_if(Notes, ""))%>% # Replace empty string notes with NAs
-  left_join(YBFMP_stations, by="Station")%>%
-  distinct(Station, Date, Datetime, .keep_all = TRUE)%>%
-  select(Source, Station, Latitude, Longitude, Date, Datetime, Tide, Microcystis, Secchi, Temperature, Conductivity, Notes)
+YBFMP_zoop <-
+  read_csv(
+    file.path(tempdir(), "Zoop_data_20211205.csv"),
+    col_types = cols_only(
+      Date = "c",
+      Datetime = "c",
+      StationCode = "c",
+      Tide = "c",
+      WaterTemperature = "d",
+      Secchi = "d",
+      Conductivity = "d",
+      SpCnd = "d",
+      MicrocystisVisualRank = "d",
+      FieldComments = "c"
+    )
+  ) %>%
+  transmute(
+    Date = ymd(Date, tz = "America/Los_Angeles"),
+    Datetime = ymd_hms(Datetime, tz = "America/Los_Angeles"),
+    Station = StationCode,
+    Tide = recode(Tide, High="High Slack", Low="Low Slack"),
+    Temperature = WaterTemperature,
+    # Convert Secchi to cm
+    Secchi = Secchi * 100,
+    # Convert Conductivity to Specific Conductance, see suisun.R for info
+    Conductivity = if_else(
+      is.na(SpCnd),
+      Conductivity / (1 + 0.019 * (Temperature - 25)),
+      SpCnd
+    ),
+    Microcystis = MicrocystisVisualRank,
+    Notes = FieldComments
+  ) %>%
+  # Remove records where all parameters are NA
+  filter(!if_all(where(is.numeric), is.na)) %>%
+  # Remove duplicated records across all columns
+  distinct()
+
+YBFMP_fish <-
+  read_csv(
+    file.path(tempdir(), "event.csv"),
+    col_types = cols_only(
+      Datetime = "c",
+      SampleDate = "c",
+      StationCode = "c",
+      WaterTemp = "d",
+      Secchi = "d",
+      Conductivity = "d",
+      SpecificConductance = "d",
+      Tide = "c",
+      MicrocystisRank = "d",
+      FieldComments = "c"
+    )
+  ) %>%
+  transmute(
+    Date = mdy(SampleDate, tz = "America/Los_Angeles"),
+    Datetime = mdy_hm(Datetime, tz = "America/Los_Angeles"),
+    Station = StationCode,
+    Tide = recode(Tide, High="High Slack", Low="Low Slack"),
+    Temperature = WaterTemp,
+    # Convert Secchi to cm
+    Secchi = Secchi * 100,
+    # Convert Conductivity to Specific Conductance, see suisun.R for info
+    Conductivity = if_else(
+      is.na(SpecificConductance),
+      Conductivity / (1 + 0.019 * (Temperature - 25)),
+      SpecificConductance
+    ),
+    Microcystis = MicrocystisRank,
+    Notes = FieldComments
+  ) %>%
+  # Remove records where all parameters are NA
+  filter(!if_all(where(is.numeric), is.na)) %>%
+  # Remove duplicated records across all columns
+  distinct()
+
+# Pull out duplicated records with identical Station-Datetime combinations from fish data
```

Comment:
Did you find out why there was so much duplicated data? The new data format should theoretically have removed any data duplication.

Reply:
Yeah, that's what I was thinking going into this as well, but there were so many records with repeated WQ values in the event.csv table. I ended up asking Nicole Kwan about this, and she was pretty certain that these repeated WQ values were from field staff checking their traps/nets multiple times at a station in one day, but only taking one set of WQ measurements at the beginning of the day. She suspected that they were duplicated in the event.csv table so that each eventID would have WQ measurements associated with them. Sometimes, it looked like field crews collected two sets of water quality measurements at the same site and day. All of this is misleading since from the event.csv table it looks like they collected multiple WQ measurements throughout the day at one station, but in reality they only collected one or sometimes two sets of measurements. The repeated records served as placeholders and did not represent an independent set of WQ measurements collected at the indicated datetime stamp.

We decided to fix this by only keeping the records with the earliest Datetime for the groups of records that shared identical water quality measurements at the same site and day. I told Nicole that for the next publication it would be best to pull out the WQ measurements from the event.csv table and put them in a separate table at a different hierarchy in the database to eliminate the repeated records. I offered to help them with this effort.

Reply:
Awesome, thanks Dave!